### PR TITLE
Ciena show vlan

### PR DESCRIPTION
**Pull request by friday963**
##### ISSUE TYPE

 - New Template Pull Request

##### COMPONENT

ciena_6x_vlan_show, ciena_6x, vlan_show

##### SUMMARY

Adding vlan show command template.

```

```

### CODE REVIEW

File: templates/ciena_6x_vlan_show.textfsm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Value VLAN_ID (\d+)
+Value VLAN_NAME ([0-9a-zA-Z_#]+)
```
Comment (FragmentedPacket) [inline templates/ciena_6x_vlan_show.textfsm:2]:
```suggestion
Value NAME ([0-9a-zA-Z_#]+)
```

This will match `cisco_ios_show_vlan.textfsm` value names.

**Comment by FragmentedPacket** [comment]
@friday963 Do you mind addressing Jacob's feedback? We did also just merge a PR that changes the name from `ciena_6x` to `ciena_soas` to match Netmiko.

File: templates/ciena_6x_vlan_show.textfsm (hunk 1 above)
Comment (friday963) [reply templates/ciena_6x_vlan_show.textfsm:2]: I've made the change in my template and will push that with NAME.

**Comment by friday963** [comment]
I just pushed this back up to my fork and I see the change in the commit history but I am unsure if there is anything to resubmit this for your review.

**Comment by FragmentedPacket** [comment]
Do you mind fixing the conflicts? And then I think we're good to go.

Do you mind renaming the other files you moved to the new dir `software_show` as well?

**Comment by friday963** [comment]
I'm sorry after I pulled down the most recent changes I'm having trouble keeping up with whats even going on.  I see files in /tests/ciena_saos/vlan_show/*.raw and *.yml and even a textfsm template that are not my mine but look to be in the master branch already.  I really am not sure what to do at this point.

**Comment by FragmentedPacket** [comment]
@friday963 Np. I will try and recreate the steps and then send them to you. I may just fix it myself as well and push up new PR, but I will let you know. I will try and get that done by the end of this weekend.

**Comment by friday963** [comment]
I'm fine with whatever you choose, thanks for your patience.  This was my
first time trying to collaborate on a project.

Respectfully,
Andrew

**Comment by FragmentedPacket** [comment]
@friday963 I looked at this and it appears `ciena_saos_vlan_show.textfsm` was merged a few days ago. Does that template work for your output?

**Comment by friday963** [comment]
No the template was failing against my raw output.  However I tested my template against the raw output from the current ciena_saos_vlan_show.raw and it was successful. So it would seem mine is able to handle more variations in raw text.  If possible I'd like to see this through to the end and actually do a successful PR.  Can we close this PR and I'll attempt this again and just modify the current template?

**Comment by FragmentedPacket** [comment]
@friday963 Sounds good. We're available on our public slack if you want to join as well.

I'll work on getting this fixed up, and then provide you the steps to do on your own and then push up to this PR.

**Comment by FragmentedPacket** [comment]
1. `git checkout master`
    1. This is needed to pull in the latest changes from upstream master
2. `git remote add upstream <https/ssh url for github.com/networktocode/ntc-tempaltes.git>`
    1. Added the ntc repo as an upstream remote in your local repo
3. `git pull upstream master --rebase`
    1. This will pull in the latest changes from NTC templates and then automatically rebase it for you so it only pulls in those commits and messages without adding the `Merged master blah blah blah`
4. `git checkout ciena_show_vlan`
    1. Get back into your original branch you used to make this PR
5. `git merge master`
    1. Copy both `tests/ciena_saos/vlan_show/ciena_saos_vlan_show.yml` and `tests/ciena_saos/vlan_show/ciena_saos_vlan_show.raw` into the same folder, but add 2 to the end of the `vlan_show` portion of the file name. e.g. `tests/ciena_saos/vlan_show/ciena_saos_vlan_show2.raw` and `tests/ciena_saos/vlan_show/ciena_saos_vlan_show.yml`.
    2. Fix the conflicts in all four of those files. The key is the keep the original data in the originally named test files and then your changes in the ones that have 2 on the end of it.
    1. Fix all the conflicts with the messages shown (I would set the current tests files back to what they were as we'll want to add our own with different files names)
        1. such as `template/index` and the files above
6. Remove `tests/ciena_saos/software_show/ciena_6x_software_show.raw` and `tests/ciena_saos/software_show/ciena_6x_software_show.yml`
    1. These are duplicates from your branch and the correctly named files were pulled in on merge. e.g. `tests/ciena_saos/software_show/ciena_saos_software_show.raw`
7. `git checkout master .gitignore`
8. `rm -rf .vscode/settings.json`
9. `git add -A`
10. `git commit`
11. `git push origin ciena_show_vlan`
    1. Push your changes and cleanup back into the PR


If you have issues with the merge, just type `git merge --abort` and hit me up on our public Slack.

**Comment by georgesnow** [comment]
@friday963 let me know if you run into issues. I was the one that made the PR for vlan show. I had made modifications cause your template failed on my raw output.  I thought I had ran it against your raw output in tests, but maybe not. So it appears the variations in output need to be addressed. @FragmentedPacket how do you prefer handle that?

**Comment by georgesnow** [comment]
@friday963 I just re-tested your template on our raw output and it fails. I tested the template I made against your raw output and it works. However, it does not take care of all the items you want to capture.

**Comment by friday963** [comment]
@georgesnow Actually what you're saying is correct.  After I had gotten my original request to fix an issue with the error state ^. -> Error I went to the slack channel and spoke with them over there.  They had helped me fix the template.  So now it works.  But you are right mine would fail against yours.

**Comment by FragmentedPacket** [comment]
Once everything is fixed in the PR (template/index names, etc.), tox can be ran and since the original data that @georgesnow included in the initial PR will show any issues that arise with the change in template.

That's why I was requesting to create your own `.raw` and `.yml` so we can always make sure to keep backwards compatibility.

**Comment by friday963** [comment]
Sorry to leave this hanging still I thought we were good.  What do I need to fix to get this finished? @FragmentedPacket

File: templates/ciena_soas_vlan_show.textfsm [hunk 2]
```diff
@@ -0,0 +1,31 @@
+Value VLAN_ID (\d+)
```
Comment (FragmentedPacket) [inline templates/ciena_soas_vlan_show.textfsm:1]: Is this template named wrong now? It shows that the correct naming template was deleted. Rename back to original name please.

**Comment by FragmentedPacket** [comment]
Closed via #707

**Pull request closed without merging**